### ISSUE TITLE
Create a Librarian and a Teacher

### ISSUE THREAD

**Issue by zombieslayer9er** [zombieslayer9er/The-Council#3]
Add Librarian weight proposals, versioned profiles, and Teacher validation

Goal

Introduce the first adaptive-learning layer without allowing learning components to modify the deterministic trading core directly.

Implement three major concepts:

1. Librarian
2. versioned weight/profile store
3. Teacher/Evaluator

The Librarian proposes bounded changes.

The Teacher decides whether those changes survive.

Hard architectural rule

The Librarian may inspect ground-truth historical outcomes and Judge results.

Specialist agents and the Council must never gain access to that ground truth during the original historical decision.

Learning happens only after an episode has been completed and judged.

Librarian responsibilities

The Librarian should analyze cached historical episodes and determine how well individual specialists performed under different conditions.

Initial analysis should support dimensions such as:

- global performance
- asset class
- individual symbol
- market regime
- forecast horizon
- recent versus long-term performance

The Librarian should produce a proposal, not directly mutate active weights.

Example concept:

agent: historical_recurrence
scope:
  symbol: SPY
  regime: bull_high_volatility
current_weight: 0.62
proposed_weight: 0.67
sample_count: 143
confidence: 0.81
reason: historically well calibrated under this regime

Slow and fast weights

Design weight profiles to support both:

- slow/long-term competence
- fast/recent adjustment

Avoid allowing short-term outcomes to overwrite long-term learned competence.

The effective policy may combine:

- global prior
- asset adjustment
- symbol adjustment
- regime adjustment
- horizon adjustment
- recent-performance adjustment

Keep the initial implementation conservative and bounded.

Weight updates

Every accepted profile must have a version/generation ID.

For example:

"weights-v0001"

"weights-v0002"

Every backtest/episode must record which generation it used.

Weight changes must be bounded by configurable maximum deltas.

Do not allow one small batch of observations to radically change a specialist's authority.

Use sample count and confidence when determining allowed update magnitude.

Teacher/Evaluator responsibilities

The Teacher receives:

- an existing weight profile
- a Librarian proposal
- cached historical episodes

It then evaluates the proposed profile using held-out data.

Critically:

Do NOT train a proposed update using Episode A and then declare it successful because it improves Episode A.

Use temporal/held-out validation.

Initial implementation should support walk-forward evaluation or another explicitly time-separated validation strategy.

Evaluation criteria

Do not optimize only raw return.

Compare old versus proposed weights using a basket of metrics such as:

- total return
- benchmark-relative return
- maximum drawdown
- hit rate
- confidence calibration
- risk-adjusted performance
- turnover/cost sensitivity
- stability across validation periods

Keep the scoring function explicit, deterministic, versioned, and testable.

Acceptance

Teacher output should clearly indicate:

- ACCEPT
- REJECT
- optionally ACCEPT_REDUCED_UPDATE

Include old/new metric comparisons and reasons.

Only an accepted result may create the next weight generation.

Historical robustness

A proposed improvement should not be considered successful merely because it improves the most recent regime while catastrophically damaging older regimes.

Include validation across multiple temporal slices where sufficient data exists.

Replay efficiency

Use frozen cached specialist outputs whenever the proposed change only affects Council weighting.

Avoid rerunning expensive specialist analysis unnecessarily.

Safety/invariants

The learning layer must not alter:

- causal timestamps
- source market data
- Judge truth
- execution semantics
- hard risk limits
- risk-governor veto authority
- historical outcomes

Learnable policy and system invariants must remain separate.

Tests

Add tests for:

- versioned weight generations
- bounded updates
- no direct Librarian mutation
- held-out validation
- rejection of overfit proposals
- deterministic Teacher decisions
- rollback to previous generation
- replay using cached specialist outputs
- ground-truth isolation
- prevention of future-data leakage

Documentation

Document the full loop:

Historical run → Judge → Experience Store → Librarian → Proposed Profile → Teacher Replay → Accept/Reject → New Weight Generation

Keep UI work outside this branch.
